src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -895,6 +909,37 @@ ModifyQuerySupported(Query *queryTree, Query *originalQuery, bool multiShardQuer
 }
 
 
+/*
+ * Modify statements on simple updetable views are not supported yet.
+ * Actually, we need the original query (the query before postgres
+ * pg_rewrite_query) to detect if the view sitting in rtable is to
+ * be updated or just to be used in FROM clause.
+ * Hence, tracing the postgres source code, we deduced that postgres
+ * puts the relation to be modified to the first entry of rtable.
+ * If first element of the range table list is a simple updatable
+ * view and this view is not coming from FROM clause (inFromCl = False),
+ * then update is run "on" that view.
+ */
+static DeferredErrorMessage *
+DeferErrorIfModifyView(Query *queryTree)
+{
+	RangeTblEntry *firstRangeTableElement = (RangeTblEntry *) linitial(
```

Comment:
Actually, it does not sound possible to me but I could not find any useful information in postgres code. It would be better to add a `queryTree->rtable == NIL` check to this function.